File: app/src/main/java/com/example/shoppingapp/fragments/HomeScreen.java
package com.example.shoppingapp.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.shoppingapp.R;
import com.example.shoppingapp.adapters.CategoryAdapter;
import com.example.shoppingapp.adapters.PreviousOrderAdapter;


public class HomeScreen extends Fragment {

    RecyclerView rv_category;
    RecyclerView rv_previous_order;
    TextView txt_see_all_category;
    TextView txt_see_all_previous_order;


    CategoryAdapter categoryAdapter;
    PreviousOrderAdapter previousOrderAdapter;
    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View v=inflater.inflate(R.layout.fragment_home_screen,container,false);
        return v;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        setUI();
    }

    private void setUI() {
        BindView();

        RecyclerView.LayoutManager rv_category_lm=new LinearLayoutManager(getContext(), LinearLayoutManager.HORIZONTAL, false);

        categoryAdapter=new CategoryAdapter(getContext());
        rv_category.setHasFixedSize(true);
        rv_category.setLayoutManager(rv_category_lm);
        rv_category.setAdapter(categoryAdapter);

        RecyclerView.LayoutManager rv_previous_order_lm=new LinearLayoutManager(getContext(), LinearLayoutManager.HORIZONTAL, false);
        previousOrderAdapter=new PreviousOrderAdapter(getContext());
        rv_previous_order.setHasFixedSize(true);
        rv_previous_order.setLayoutManager(rv_previous_order_lm);
        rv_previous_order.setAdapter(previousOrderAdapter);
    }
    
    void BindView(){
         rv_category=getView().findViewById(R.id.category_recycler_view);
         rv_previous_order   =getView().findViewById(R.id.previous_order_recycler_view);
         txt_see_all_category  =getView().findViewById(R.id.txt_see_all_category);
         txt_see_all_previous_order  =getView().findViewById(R.id.txt_see_all_previous_order);
    }
}

File: app/src/customer/java/com/example/shoppingapp_customer/fragments/HomeScreen.kt
package com.example.shoppingapp_customer.fragments

import com.example.shoppingapp_customer.adapters.CategoryAdapter
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.view.View
import android.widget.GridLayout
import androidx.fragment.app.Fragment
import androidx.navigation.findNavController
import androidx.recyclerview.widget.GridLayoutManager
import com.example.shoppingapp.R
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.shoppingapp.Item
import com.example.shoppingapp.User
import com.example.shoppingapp.util.Firebase
import kotlinx.android.synthetic.customer.fragment_home_screen.*

class HomeScreen : Fragment() {

    private lateinit var categoryAdapter:CategoryAdapter

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_home_screen, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        fab_logout.setOnClickListener{
            Firebase.logout(activity = requireActivity())
        }
        setUI()
        img_cart_hsf.setOnClickListener {
            it.findNavController().navigate(R.id.action_homeScreen_to_cartScreen)
        }
    }

    private fun setUI() {
        categoryAdapter = CategoryAdapter(requireContext())
        rv_category_hsf!!.setHasFixedSize(true)
        rv_category_hsf.layoutManager = GridLayoutManager(requireContext(), 2)
        rv_category_hsf.adapter = categoryAdapter
    }

}